src/ol/source/tilejsonsource.js

```diff
@@ -62,6 +56,36 @@ goog.inherits(ol.source.TileJSON, ol.source.TileImage);
 
 
 /**
+ * @private
+ * @param {Event} event The load event.
+ */
+ol.source.TileJSON.prototype.onXHRLoad_ = function(event) {
+  var client = /** @type {XMLHttpRequest} */ (event.target);
+  if (client.status >= 200 && client.status < 300) {
+    var response;
+    try {
+      response = /** @type {TileJSON} */(JSON.parse(client.responseText));
+    } catch (err) {
+      this.handleTileJSONError();
+      return;
+    }
+    this.handleTileJSONResponse(response);
+  } else {
+    this.handleTileJSONError();
+  }
+};
+
+
+/**
+ * @private
+ * @param {Event} event The error event.
+ */
+ol.source.TileJSON.prototype.onXHRError_ = function(event) {
+  this.handleTileJSONError();
+};
```

Comment:
Well, this gets called with a `XMLHttpRequestProgressEvent`, which the `handleTileJSONError` method doesn't expect.  We currently ignore this event, but it is the source of all the useful information, and it could make sense to make it available somehow.  But we also have things like parsing errors that result in `handleTileJSONError`, so it doesn't make sense that this method should always expect a `XMLHttpRequestProgressEvent`.